src/libcore/mem.rs

```diff
@@ -1131,6 +1151,35 @@ impl<T> MaybeUninit<T> {
     ///
     /// Note that dropping a `MaybeUninit` will never call `T`'s drop code.
     /// It is your responsibility to make sure `T` gets dropped if it got initialized.
+    ///
+    /// # Example
+    ///
+    /// Correct usage of this method: initializing a struct with zero, where all
+    /// fields of the struct can hold 0 as a valid value.
```

Comment:
```suggestion
    /// fields of the struct can hold the bit-pattern 0 as a valid value.
```
(to clarify that this is not some semantic "zero")